app/services/foreman_ansible/playbooks_importer.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+module ForemanAnsible
+  # Imports playbooks from smart proxy
+  class PlaybooksImporter
+    include ::ForemanAnsible::ProxyAPI
+    delegate :playbooks, :playbooks_names, :to => :proxy_api
+
+    def initialize(proxy = nil)
+      @ansible_proxy = proxy
+    end
+
+    def import_playbooks(playbooks_names)
+      import_playbooks_content playbooks(playbooks_names)
+    end
+
+    def import_playbooks_content(playbooks)
+      params = { changed: { update: [], create: [] } }
+      playbooks.each do |playbook|
+        new_playbook = parse_playbook playbook
+        job_template_id = JobTemplate.where(name: new_playbook[:name]).pluck(:id).first
+        old_var_block = PlaybooksImporterHelper.check_vars_block new_playbook[:old_vars_block]
+        if job_template_id.present?
+          params[:changed][:update].push(
+            {
+              job_template_id: job_template_id,
+              playbook: new_playbook,
+              old_var_block: old_var_block
+            }
+          )
+        else
+          params[:changed][:create].push(
+            {
+              playbook: new_playbook,
+              old_var_block: old_var_block
+            }
+          )
+        end
+      end
+      params
+    end
+
+    def parse_playbook(playbook)
+      playbook_name =  playbook['name']
+      metadata = add_metadata playbook['name']
+      new_playbook_content = convert_to_job_template_input playbook['playbooks_content']
+      {
+        name: playbook_name,
+        playbook_content: metadata + new_playbook_content[:playbook_content],
+        template_inputs: new_playbook_content[:template_inputs],
+        old_vars_block: new_playbook_content[:old_vars_block]
+      }
+    end
+
+    def add_metadata(playbook_name)
+      [
+        '<%#',
+        "  name: #{playbook_name}",
+        '  snippet: false',
+        '  job_category: Ansible Playbook - Imported',
+        '  provider_type: Ansible',
+        '  kind: job_template',
+        '  model: JobTemplate',
+        "%>\n"
+      ].join("\n")
+    end
+
+    def convert_to_job_template_input(playbook_content)
+      playbook_content_as_json = YAML.safe_load(playbook_content.join('')).first
```

Comment:
This looks like we have an input format for playbook_content to be rows strings with the new lines in an array. Why? Why don't we have the full string instead as that would feel like more standardized way to pass that arround (especially over the wire)?

Reply:
This is the way the playbooks are stored in smart proxy:
```
{"name"=>"xprazak2.forklift_collection.collect_debug_draft.yml",
 "playbooks_content"=>
  ["---\n",
   "- hosts: all\n",
   "  become: true\n",
   "  vars:\n",
   "    bats_output_dir: '/root/bats_results'\n",
   "    remote_dir: \"/tmp/debug-{{ pipeline_type | default('foreman') }}-{{ pipeline_version | default('nightly') }}-{{ pipeline_os | default('el7') }}\"\n",
   "  roles:\n",
   "    - sos_report\n",
   "  tasks:\n",
   "    - name: \"Find bats files\"\n",
   "      find:\n",
   "        paths: \"{{ bats_output_dir }}\"\n",
   "        patterns: \"*.tap\"\n",
   "      register: bats_results\n",
   "\n",
   "    - name: \"Copy bats results\"\n",
   "      fetch:\n",
   "        src: \"{{ item.path }}\"\n",
   "        dest: \"{{ remote_dir }}\"\n",
   "      with_items: \"{{ bats_results.files }}\"\n",
   "\n",
   "    - name: \"Find smoker files\"\n",
   "      find:\n",
   "        paths: \"{{ smoker_output_dir }}\"\n",
   "        patterns: \"*.xml\"\n",
   "      register: smoker_results\n",
   "\n",
   "    - name: \"Copy smoker results\"\n",
   "      fetch:\n",
   "        src: \"{{ item.path }}\"\n",
   "        dest: \"{{ remote_dir }}\"\n",
   "      with_items: \"{{ smoker_results.files }}\"\n"]}
```

In proxy we get playbooks_content from:
```
   playbooks_content: File.readlines(path)
   ```
   Do you think it should be a different way?

Reply:
yeah, I believe it would be better to send it as output of `File.read(path)` instead if possible (just because it is more reliable to send strings then arrays and we are joining it all together on the foreman end anyway).
It is not a major issue tho.

Reply:
so I'll move the join part into proxy in a different PR?